admin-dev/themes/new-theme/js/pages/feature-flag/index.js

```diff
@@ -30,10 +30,22 @@ const {$} = window;
 $(() => {
   const $submitButton = $('#submit-btn-feature-flag');
   const $form = $('#feature-flag-form');
+  const initialState = $form.serialize();
 
   $submitButton.on('click', (event) => {
     event.preventDefault();
 
+    const formData = $form.serializeArray();
+    let oneFlagIsEnabled = false;
+    const noFieldModifiedInTheForm = (initialState === $form.serialize());
+
```

Comment:
```suggestion
if (noFieldModifiedInTheForm) {
   $form.submit();
   return;
}
```

Reply:
That is true. Smart 👍

Reply:
Do we really need to submit if no fields are modified?

Reply:
I would say yes.
Image following scenario: you open 2 tabs of the same page.

On tab 1 you modify values, you hit "Save!" it changes the values.
You want to reset values.
You switch to tab 2 that has no modified fields and kept the initial values. You hit "save".

I think this Javascript logic should only control "when or not to open the confirmation modal", not "when or not submit the form" 😉

Reply:
I don't really agree, let's say you had a feature activated, now on tab 1 you disable it and save, then on tab 2 you hit save: no modal. I think you should have had a modal.
Maybe we should disable the "save" button when no change is made on the form rather than send the form

Reply:
> I don't really agree, let's say you had a feature activated, now on tab 1 you disable it and save, then on tab 2 you hit save: no modal. I think you should have had a modal.

That is another topic 🤔 (but we can discuss it).

> Maybe we should disable the "save" button when no change is made on the form rather than send the form

I disagree 😄 . I dont think it's the JavaScript responsibility to decide "yes or no we submit". I think the JavaScript only responsibility should be "open the confirm modal or not".

Also if we disable the "save" button when no change is made on the form, it means the page does not work when JavaScript is disabled 🤔 . But let's be honest: if user disables the use of JavaScript on the Back Office a lot of things are blocked.

> Maybe we should disable the "save" button when no change is made on the form rather than send the form

@atomiix do you want to ask the PM about it? you have your opinion, I have mine. But it's a behavior topic so it's supposed to be for the PM ?

Reply:
@matks let's ask @PrestaShop/product-team :)

Reply:
I agree with @atomiix :roll_eyes:

Reply:
I agree with @PierreRambaud who is agreed with @atomiix, it will be easier. Worst case scenario, the merchant enable the experimental feature again, and it avoids having a modal every time :)

Reply:
Actually it is cool behavior, that will be interesting to add for all the form :)

Reply:
So if I understand this PR's feature is now to enable/disable the save button when the data is changed?
That doesn't change the fact that the modal is displayed on more specific conditions (only when a flag has just been enabled).

Reply:
Indeed @jolelievre I updated the issue https://github.com/PrestaShop/PrestaShop/issues/24267 to acknowledge the new behavior
- something is changed in the form: we show the modal IF a feature flag has been enabled
- nothing has changed in the form (user did not modify it or restored initial state): button SAVE is disabled and consequently modal is not opened on click